src/ArrOperation.java

```diff
@@ -0,0 +1,152 @@
+public class ArrOperation
+{
+    private int array[];
+    // Parameterized constructor
+    public ArrOperation(int array[])
+    {
+        if(array.length>0)
+        {
+            this.array = array;
+        }
+        else
+        {
+            System.out.println("Can not perform operations on null array");
+            System.exit(-1);
+        }
```

Comment:
1. Move this validation part into corresponding methods. 
2. And throw assertion error in case the array is empty from there.
3. Verify these case in test method

Reply:
Verified using catch assertion error Junit runner